### PR TITLE
Redirect postgres log messages to spdlog

### PR DESCRIPTION
**Pull request by lebdron**
### Description of the Change
Print notices from postgres to spdlog instead of std::cerr

### Benefits
Logging is controlled consistently with other logs

### Possible Drawbacks 
`reinterpret_cast`...

### CODE REVIEW

File: irohad/ametsuchi/impl/storage_impl.cpp [hunk 1]
```diff
@@ -44,6 +45,17 @@ namespace {
     }
   }
 
+  std::string formatPostgresMessage(const char *message) {
+    std::string formatted_message(message);
+    boost::replace_all(formatted_message, "\n", " ");
```
Comment (MBoldyrev) [inline irohad/ametsuchi/impl/storage_impl.cpp]: maybe replace "\r" too?

File: irohad/ametsuchi/impl/storage_impl.cpp [hunk 2]
```diff
@@ -44,6 +45,17 @@ namespace {
...
+    return formatted_message;
+  }
+
+  void processPqNotice(void *arg, const char *message) {
+    auto *log = reinterpret_cast<logger::Logger *>(arg);
```
Comment (igor-egorov) [inline irohad/ametsuchi/impl/storage_impl.cpp:56]: Please check if dynamic_cast can be employed here instead of reinterpret_cast.

Comment (MBoldyrev) [reply irohad/ametsuchi/impl/storage_impl.cpp:56]: no, it cannot be employed with `void*`

**Comment by lebdron** [comment]
Linux CI has passed, Mac CI has a known issue.

**Pull request merged**